server/mergin/tests/test_auth.py

```diff
@@ -889,3 +885,31 @@ def test_server_usage(client):
     assert resp.json["editors"] == 1
     assert resp.json["users"] == 1
     assert resp.json["projects"] == 1
+
+
+user_data = [
+    ("user1", 201),  # no problem
+    ("user\260", 400),  # disallowed character
+    ("usér", 201),  # non-ascii character
+    ("user\\", 400),  # disallowed character
+    ("日人日本人", 201),  # non-ascii character
+    ("user|", 400),  # vertical bar
+    ("us er", 400),  # whitespace
+    ("us,er", 400),  # comma
+    ("us—er", 400),  # dash
+    ("us'er", 400),  # apostrophe
```

Comment:
how about these use cases:
- whitespace around word
- starting with .
- using . in word